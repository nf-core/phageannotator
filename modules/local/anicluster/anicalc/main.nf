process ANICLUSTER_ANICALC {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' :
        'biocontainers/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' }"

    input:
    tuple val(meta), path(blast_txt)

    output:
    tuple val(meta), path("*_ani.tsv")  , emit: ani
    path "versions.yml"                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    anicalc.py \\
        -i $blast_txt \\
        -o ${prefix}_ani.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(echo \$(biopython_version.py 2>&1))
        numpy: \$(echo \$(numpy_version.py 2>&1))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_ani.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(echo \$(biopython_version.py 2>&1))
        numpy: \$(echo \$(numpy_version.py 2>&1))
    END_VERSIONS
    """
}
