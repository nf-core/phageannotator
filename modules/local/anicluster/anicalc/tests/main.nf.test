nextflow_process {

    name "Test process: ANICLUSTER_ANICALC"
    script "../main.nf"
    process "ANICLUSTER_ANICALC"


    test("blast.txt") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.pipelines_testdata_base_path + 'modules/local/anicluster/anicalc/blast.txt', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("blast.txt - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.pipelines_testdata_base_path + 'modules/local/anicluster/anicalc/blast.txt', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
