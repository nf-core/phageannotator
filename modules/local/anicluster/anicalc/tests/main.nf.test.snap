{
    "blast.txt - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_ani.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e762f9268614324a9143737b92563257"
                ],
                "ani": [
                    [
                        {
                            "id": "test"
                        },
                        "test_ani.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e762f9268614324a9143737b92563257"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-19T16:28:59.168124925"
    },
    "blast.txt": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_ani.tsv:md5,1934eb09849f6df88083ad25aa47923d"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e762f9268614324a9143737b92563257"
                ],
                "ani": [
                    [
                        {
                            "id": "test"
                        },
                        "test_ani.tsv:md5,1934eb09849f6df88083ad25aa47923d"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e762f9268614324a9143737b92563257"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-19T16:28:47.920248772"
    }
}
